chain/client/src/client_actor.rs

```diff
@@ -249,6 +249,11 @@ impl Actor for ClientActor {
 
         // Start catchup job.
         self.catchup(ctx);
+
+        // Tell the network what we think the current state of the chain is. If
+        // this fails, something's really wrong with the chain, so let's just
+        // crash on startup.
+        self.client.send_network_chain_info().unwrap();
```

Comment:
This is ok, however I'm very curious why this is needed.
height and tracked_shards are not interpreted by PeerManagerActor (I think) but just sent to other peers during handshake. The default value stored in PeerManagerActor is height = 0 and tracked_shards=[], which is a valid state in a sense.

Reply:
@pompon0 debugged this, this is needed to trigger chain sync if there's a network partition. 

Imagine you are a node, and you've just started up, and you have 100 blocks in the storage. Imagine also that you are partitioned from the network, so you are not receiving new blocks. In this situation, you'll end up with `height: 0` forever, although you do store 100 blocks.


Now, a new node starts up alongside you. This nodes starts with an empty storage (height 0). This node starts talking to you, and we expect it to get 100 blocks from you. But, because you advertise the height as 0, this node doesn't actually know you have 100 blocks and cant request sync from you.

Reply:
makes sense